### ISSUE TITLE
Need docs about GlobalStatementSyntax

### ISSUE THREAD

**Issue by KirillOsenkov** [dotnet/roslyn#17118]
As a user, how do I find out about the purpose of GlobalStatementSyntax? Is it documented anywhere?

**Comment by tmat** [comment]
We also need a scripting specification.

**Comment by AdamSpeight2008** [comment]
@KirillOsenkov  Check on the roslyn repo, as it was implemented back on there. It is used in vb the put the namespace into the topmost gloval namespace eg. `Global.Microsoft`

**Comment by raffaeler** [comment]
Is this used only in VB? Can someone make an example (code) representing a GlobalStatementSyntax please?

**Comment by svick** [comment]
@raffaeler  It's used in scripting. For example, in the following code, `member` will be `GlobalStatementSyntax`:

```c#
var script = CSharpScript.Create("System.Console.WriteLine();");
var member = script.GetCompilation().SyntaxTrees.Single().GetCompilationUnitRoot().Members.Single();
```

Though unlike when this issue was created, there is now [an empty documentation page for `GlobalStatementSyntax`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.csharp.syntax.globalstatementsyntax). Maybe this issue should be moved to [the roslyn-api-docs repo](https://github.com/dotnet/roslyn-api-docs/), where that page lives?

**Comment by raffaeler** [comment]
Thank you @svick 
The whole documentation looks like being autogenerated and totally useless.

There is no snippet in any of the syntax doc pages. Some syntax kind are straightforward (declaration and similar) but many others are not obvious at all.
At the moment I created a snippet for each of them, and GlobalStatementSyntax was one of the last missing in my own doc, so thank you again.

**Comment by svick** [comment]
@raffaeler Yeah, those docs are definitely a work in progress. It's also open to contributions, so if you would like to make improvements to it (possibly based on "your own doc"), I'm sure it would be appreciated.

**Comment by raffaeler** [comment]
Sure. But my priority is now to finish filling the gaps, with a snippet for each of them:
- `ShebangDirectiveTriviaSyntax` (scripting, I guess)
- `AttributeTargetSpecifierSyntax` (obviously attribute related, but never seen that in the visualizer)
- `XmlProcessingInstructionSyntax` (this looks strange as I won't expect the processing instruction inside the comments)
- `ConversionOperatorMemberCrefSyntax` (I tried specifying the cref for conversion operators, but it wasn't used)
- `TypeCrefSyntax`
I believe I have identified all the others.
After that, I will have to format appropriately the doc to highight the relevant part of the syntax (which looks trivial, but the list is super-long).
